server/src/main/java/io/crate/execution/engine/collect/files/LineParser.java

```diff
@@ -52,6 +52,9 @@ private enum InputType {
     public void readFirstLine(URI currentUri,
                               FileUriCollectPhase.InputFormat inputFormat,
                               BufferedReader currentReader) throws IOException {
+        for (int i = 0; i < parserProperties.skipNumLines(); i++) {
+            currentReader.readLine();
+        }
```

Comment:
Shouldn't this happen within the `FileReadingIterator` to ensure `currentLineNumber` gets advanced correctly? Or does the retry mechanism work despite that?

And I think if `skipNumLines` is a long, `i` would need to be too:

```suggestion
        for (long i = 0; i < parserProperties.skipNumLines(); i++) {
            currentReader.readLine();
        }
```

Reply:
Thanks. updated it to `long` and  `currentLineNumber` gets advanced correctly, added a test to confirm that.